pkg/controller/elasticsearch/driver/esstate.go

```diff
@@ -115,21 +117,26 @@ func (s *memoizingShardsAllocationEnabled) ShardAllocationsEnabled() (bool, erro
 
 // memoizingHealth provides cluster health information.
 type memoizingHealth struct {
-	health   esv1.ElasticsearchHealth
-	once     sync.Once
-	esClient esclient.Client
-	ctx      context.Context
+	health     esv1.ElasticsearchHealth
+	safeToRoll bool
+	once       sync.Once
+	esClient   esclient.Client
+	ctx        context.Context
 }
 
 // initialize requests Elasticsearch for cluster health, only once.
 func (h *memoizingHealth) initialize() error {
 	ctx, cancel := context.WithTimeout(h.ctx, esclient.DefaultReqTimeout)
 	defer cancel()
-	health, err := h.esClient.GetClusterHealth(ctx)
+
+	// get cluster health but make sure we have no pending shard initializations
+	// by requiring the event queue to be empty
```

Comment:
> by requiring the event queue to be empty

This is slightly misleading as I would think we return an error here if the event queue is not empty.
Maybe:

```
// get cluster health and status of the event queue (are there pending events?)
```